StevesSixLeggedSprinters/BangazonAPI
# Pkdotnet

## Status
Choose:
**HOLD**

## Migrations
YES

## Description
I have created an Order model (Order.cs) in order to create a table in the database for Orders. I also created an order controller(OrderController.cs) to Get (read) list of all orders, Get a single specific order, Post(create) an order, Put(update) a specific order, and Delete a specific order. I have also modified BangazonApIContext file to set up a relationship between the database and order table. Also created a model builder to generate time-stamp for when an order is created. 

I also made following changes -
1. Changed Icollection customer to order to show multiple orders in the collection.
2. Corrected the spelling on DateOrderd to DateOrderedIn in Order.cs line 19. 


## Todos


## Deploy Notes
Not deployed


## Steps to Test or Reproduce
Outline the steps to test or reproduce the PR here.

```sh
I have set following methods: GET, POST, PUT, DELETE

Follow below steps to test

git fetch --a  (fetch all branches from GitHub)
git branch -avv (display all branches)
git checkout pkdotnet (switch into my branch)
git pull origin pkdotnet (pull my code down to test)
dotnet ef migrations add Initial (migrations need to be built)
dotnet ef database update (update the database since a new migration has been built)
open bangazon.db (open database)
dotnet run

Make sure to have Postman installed: https://www.getpostman.com/
In Postman, 'Post' (add) few sample order before testing.
Also, make sure http://localhost:5000/ is running

Example:
{
    "PayTypeId": 4,
	"DateOrdered": "07/27/2017 9:45:06 PM",
	"CustomerId": 1,
	"Customer": "Preeti"
}

To view all products:
Select GET
http://localhost:5000/api/order

To view a single product:
Select GET
http://localhost:5000/api/order/{id}

To edit a single product
Select PUT
http://localhost:5000/api/order/{id}

To delete a single product
Select DELETE
http://localhost:5000/api/order/{id}

```

## Impacted Areas in Application
List general components of the application that this PR will affect:

* Models\Order.cs
* Controllers\OrderController.cs
